src/array_api_stubs/_draft/elementwise_functions.py

```diff
@@ -1339,56 +1383,60 @@ def floor_divide(x1: array, x2: array, /) -> array:
-
     Parameters
     ----------
-    x1: array
+    x1: Union[array, int, float]
         first input array. Should have a real-valued data type.
-    x2: array
+    x2: Union[array, int, float]
         second input array. Must be compatible with ``x1`` (see :ref:`broadcasting`). Should have a real-valued data type.
 
     Returns
     -------
     out: array
         an array containing the element-wise results. The returned array must have a data type of ``bool``.
 
-    .. note::
-       Comparison of arrays without a corresponding promotable data type (see :ref:`type-promotion`) is undefined and thus implementation-dependent.
+    Notes
+    -----
+
+    -   At least one of ``x1`` or ``x2`` must be an array.
+    -   Comparison of arrays without a corresponding promotable data type (see :ref:`type-promotion`) is undefined and thus implementation-dependent.
+    -   For backward compatibility, conforming implementations may support complex numbers; however, inequality comparison of complex numbers is unspecified and thus implementation-dependent (see :ref:`complex-number-ordering`).
 
     """
 
 
-def greater_equal(x1: array, x2: array, /) -> array:
+def greater_equal(
+    x1: Union[array, int, float], x2: Union[array, int, float], /
+) -> array:
     """
     Computes the truth value of ``x1_i >= x2_i`` for each element ``x1_i`` of the input array ``x1`` with the respective element ``x2_i`` of the input array ``x2``.
 
-    .. note::
-       For backward compatibility, conforming implementations may support complex numbers; however, inequality comparison of complex numbers is unspecified and thus implementation-dependent (see :ref:`complex-number-ordering`).
-
     Parameters
     ----------
-    x1: array
+    x1: Union[array, int, float]
         first input array. Should have a real-valued data type.
-    x2: array
+    x2: Union[array, int, float]
         second input array. Must be compatible with ``x1`` (see :ref:`broadcasting`). Should have a real-valued data type.
 
     Returns
     -------
     out: array
         an array containing the element-wise results. The returned array must have a data type of ``bool``.
 
-    .. note::
-       Comparison of arrays without a corresponding promotable data type (see :ref:`type-promotion`) is undefined and thus implementation-dependent.
+    Notes
+    -----
+
+    -   At least one of ``x1`` or ``x2`` must be an array.
+    -   Comparison of arrays without a corresponding promotable data type (see :ref:`type-promotion`) is undefined and thus implementation-dependent.
+    -   For backward compatibility, conforming implementations may support complex numbers; however, inequality comparison of complex numbers is unspecified and thus implementation-dependent (see :ref:`complex-number-ordering`).
     """
 
 
-def hypot(x1: array, x2: array, /) -> array:
+def hypot(x1: Union[array, float], x2: Union[array, float], /) -> array:
```

Comment:
Presumably int should also be allowed here to match https://data-apis.org/array-api/latest/API_specification/type_promotion.html#mixing-arrays-with-python-scalars

> a Python int or float for real-valued floating-point array data types.

Reply:
I've matched the text, where we say that `x1` and `x2` should have a real-valued floating-point data type.

Reply:
Well this is something we should decide on. Should functions that operate on floating-point inputs accept `int`? If we are matching what we wrote for scalars with operators, then yes. If we want to be more strict, then no, but then we should probably update that section (which we need to do anyway so it applies to both operators and functions).

Reply:
`int` should be accepted everywhere I think, like the link in the first comment here.

I think this has been decided on for us already by the language; `int` is being made more and more compatible with float, and Mypy et al. have decided that `int` is a subtype of `float`: https://stackoverflow.com/questions/59619778/mypy-why-is-int-a-subtype-of-float. 

In practice it will also "just work" without changes, while trying to reject `int` inputs is cumbersome.

Reply:
I went ahead and updated to enumerate explicit `int` support; however, two points:

1. If `int` is already considered a `float`, it is not clear to me that saying `Union[int, float]` is even necessary. Simply because a type signature only includes `float` does not mean that downstream libraries would need to "reject" scalar values of type `int`. Meaning, the "just work" aspect should hold true whether or not we explicitly enumerate `int` as an accepted scalar type. Related: https://github.com/python/typing/issues/1746 and https://discuss.python.org/t/clarifying-the-float-int-complex-special-case/54018/30.
2. I think that (1) is symptomatic of a larger issue which is the sometimes conflicting need to have the specification also serve the role of an authoritative type implementation. For several of the functions, the "spirit" of the specification is that, e.g., a function is intended to work with real-valued floating-point numbers. We then muddy the waters a bit by including explicit type annotations which reflect more an implementation detail than the intent, especially when providing an `int` may result in undefined behavior due to casting.

On the one hand having types in the signatures is nice for Sphinx and API docs; on the other hand, I see advantages to more clearly separating specification guidance from a concrete type implementation.

Reply:
> especially when providing an `int` may result in undefined behavior due to casting.

I can't think of a way that this could ever happen given our casting rules which give Python scalars lowest priority.

Reply:
The undefined behavior would stem from cross-kind casting (e.g., an integer value which exceeds the maximum safe floating-point integer).